components/builder-graph/src/graph_helpers.rs

```diff
@@ -219,8 +221,14 @@ pub fn dump_graph_raw(graph: &DiGraphMap<PackageIdentIntern, EdgeType>,
                     EdgeType::BuildDep => bdeps.push(succ),
                     EdgeType::RuntimeDep => rdeps.push(succ),
                     EdgeType::StrongBuildDep => sdeps.push(succ),
+                    EdgeType::ExternalConstraint => {
+                        unimplemented!("External Constraints should not appear here")
```

Comment:
I'm curious if there is something we can do here other than insert a `panic` waiting to happen? For that reason, I don't think the pattern of utilizing `unimplemented!()`  for production code, when used in this particular manner, is going to be a good precedent to set.

Maybe we're better served if this returns an `Err` or is handled in some other non-fatal way.

From the docs:
```
Indicates unimplemented code by panicking with a message of "not implemented".

This allows your code to type-check, which is useful if you are prototyping or implementing a trait that requires multiple methods which you don't plan of using all of.
```

Where we've utilized `unimplemented!()` before was the latter case - for the trait pattern abstraction where we didn't plan on ever calling the functions but wanted the code to compile. This feels different to me since we're in a `match` and we cannot guarantee the variant won't be found, ultimately causing a service crash.

Reply:
TIL: there's also a `todo!()` macro to handle prototyping https://doc.rust-lang.org/std/macro.todo.html